src/components/MultiHopTrade/helpers.ts

```diff
@@ -85,23 +86,43 @@ export const getNetReceiveAmountCryptoPrecision = ({
   return netReceiveAmountCryptoPrecision.toString()
 }
 
+export const _getHopTotalNetworkFeeFiatPrecision = (
+  tradeQuoteStep: TradeQuote['steps'][number],
+  getFeeAssetRate: (feeAssetId: AssetId) => string,
+): BigNumber => {
+  // TODO(woodenfurniture): handle osmo swapper crazy netowrk fee logic here
+  const feeAsset = selectFeeAssetById(store.getState(), tradeQuoteStep.sellAsset.assetId)
+
+  if (feeAsset === undefined)
+    throw Error(`missing fee asset for assetId ${tradeQuoteStep.sellAsset.assetId}`)
+
+  const feeAssetFiatRate = getFeeAssetRate(feeAsset.assetId)
+
+  const networkFeeCryptoBaseUnit = tradeQuoteStep.feeData.networkFeeCryptoBaseUnit
+  const networkFeeFiatPrecision = bnOrZero(
+    fromBaseUnit(networkFeeCryptoBaseUnit, feeAsset.precision),
+  ).times(feeAssetFiatRate)
+
+  return networkFeeFiatPrecision
+}
+
+export const getHopTotalNetworkFeeFiatPrecision = (
+  tradeQuoteStep: TradeQuote['steps'][number],
+): string => {
+  const state = store.getState()
+  const getFeeAssetFiatRate = (feeAssetId: AssetId) =>
+    selectMarketDataByFilter(state, {
+      assetId: feeAssetId,
+    }).price
+  return _getHopTotalNetworkFeeFiatPrecision(tradeQuoteStep, getFeeAssetFiatRate).toString()
```

Comment:
Totally fine as-is but a bit confusing to have a `_` variant, if the goal is to inject the `getFeeAssetFiatRate` as second positional arg, could do a `withGetFeeAssetFiatRate` higher order function doing this instead?
Then `getHopTotalNetworkFeeFiatPrecision` itself would always expect the two args